### PR TITLE
Fix sass deprecation warnings and migrated to the modern sass compiler

### PR DESCRIPTION
**Pull request by The-CodeINN**
### Refactor Sass Styles to Address Deprecation Warnings and Fix Mixin Issues

This pull request refactors the Sass styles across the project to resolve deprecation warnings related to mixed declarations and fix issues with undefined mixins. The changes align the styles with upcoming Sass updates and modern CSS practices

### CODE REVIEW

File: astro.config.ts [hunk 1]
```diff
@@ -19,7 +19,6 @@ await symlink(path.resolve("content"), path.resolve("public/content"));
 export default defineConfig({
 	site: siteUrl,
 	adapter: vercel({
-		// Uses Vercel's Image Optimization API: https://vercel.com/docs/image-optimization
```
Comment (crutchcorn) [inline astro.config.ts:22]: Undo these comment deletions please 😄

Comment (The-CodeINN) [reply astro.config.ts]: Alright, will do.

File: src/styles/post-body.scss [hunk 2]
```diff
@@ -206,12 +215,3 @@
...
 }
-
-@import "src/styles/markdown/blockquote";
-@import "src/styles/markdown/codeblock";
-@import "src/styles/markdown/embed";
-@import "src/styles/markdown/hints";
-@import "src/styles/markdown/lists";
-@import "src/styles/markdown/table";
-@import "src/styles/markdown/tooltips";
-@import "src/styles/markdown/file-list";
```
Comment (crutchcorn) [inline src/styles/post-body.scss:217]: These imports need to be rearranged again to their original order. The order of the imports and CSS are important

Comment (The-CodeINN) [reply src/styles/post-body.scss:217]: So I ran into an error when I tried with the original order with the `@use` keyword
` [ERROR] [sass] @use rules must be written before any other rules.`

image

**Comment by The-CodeINN** [comment]
@crutchcorn  

I've resolved the deprecation warnings, kindly review if there are more changes to made

File: src/styles/post-body.scss (hunk 2 above)
Comment (crutchcorn) [reply src/styles/post-body.scss:217]: Dang... This change is breaking things for us:

image

Let me try something to fix this

File: src/views/base/translations/translations-header.module.scss
Comment (The-CodeINN) [inline src/views/base/translations/translations-header.module.scss:1]: @crutchcorn 

So I was reading through the original issue, and I saw that your mention that the use of `&` will be deprecated soon.

**Review by crutchcorn — APPROVED** [review]
INCREDIBLE work, thank you!!

**Comment by The-CodeINN** [comment]
> INCREDIBLE work, thank you!!

Way to go!!! It was quite fun working given the number of files 😀

**Pull request merged**